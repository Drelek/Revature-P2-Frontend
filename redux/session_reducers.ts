import { combineReducers, createStore } from "redux";
import { AppAction, IAppActions } from "./actions";
import { IAppState, initialState } from "./store";


export const reducers = (state: IAppState = initialState, action: IAppActions): IAppState => {
    const newState = { ...state };
    switch (action.type) {
        case AppAction.LOGIN:
            return { ...newState, ...action.payload };
        case AppAction.LOGOUT:
            return { ...state, user: undefined, auth: undefined };
        case AppAction.UPDATE_USER:
            return { ...state, user: action.payload?.user };
        case AppAction.TOGGLE_CANVAS:
            return { ...state, canvas: !state.canvas };
        case AppAction.TOGGLE_FEED:
            return { ...state, feed: !state.feed };
        default:
            return newState;
    }
}

