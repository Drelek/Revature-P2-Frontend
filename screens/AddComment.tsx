import * as React from 'react';
import { Text, TextInput, StyleSheet, Pressable, View } from 'react-native';
import { useState } from 'react';

const AddComment = () => {

    const[newComment, setNewComment] = useState(' ');

    const postComment = () => {
        console.log("hello")
        //TODO
        //Post comment to endpoint and refresh screen
    }

    return(
        <View style={styles.card}>
            <View style={styles.inputContainer}> 
                <TextInput
                    placeholder="Add a comment..."
                    placeholderTextColor="white" 
                    style={styles.inputBox}
                    onChangeText={(text)=> setNewComment(text)}
                />
            </View>

            <View style={styles.buttonContainer}>
                <Pressable style={styles.pressable} onPress={() => postComment()}        >
                    <Text style={styles.text}>Submit</Text>
                </Pressable>
            </View>
            
        </View>
    )
}

export default AddComment;

const styles = StyleSheet.create({
    inputBox: {
        backgroundColor: "rgb(33, 37, 41)",
        paddingVertical:9,
        marginHorizontal:5,
        fontSize:18,
        fontFamily:"Montserrat",
        textAlign:"center",
        color: "white",
    },
    card: {
        backgroundColor: 'rgb(33, 37, 41)',
        flex:1, 
        margin:0,
        padding:0,
        borderColor: 'purple',
        borderWidth: 3,
        borderTopEndRadius:10,
        borderTopStartRadius:10,
        // borderBottomEndRadius:50,
        // borderBottomStartRadius:50,
    },

    inputContainer: {
        marginBottom:10
    },

    buttonContainer: {
        // justifyContent: ''
    },

    pressable:{
        backgroundColor:"purple",
    },

    text:{
        color:"white",
        fontSize:22,
        paddingTop:5,
        paddingBottom:100,
        fontFamily:"Montserrat",
        textAlign:"center"
    }
})