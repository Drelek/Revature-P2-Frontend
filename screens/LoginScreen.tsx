import * as React from 'react';
import { StyleSheet, TextInput, SafeAreaView,  Pressable, Text, KeyboardAvoidingView } from 'react-native';
import { useState } from 'react';
import { screenWidth } from '../constants/Layout';

const LoginScreen: React.FC = () => {
  const[username, setUsername] = useState(' ');
  const[password, setPassword] = useState(' ');

  const validateLogin = () => {

  }

  return (
    <SafeAreaView style={styles.safeArea}>
      <SafeAreaView style={styles.safeArea}>
        <TextInput
        style={styles.input}
        placeholder="Username"
        placeholderTextColor="antiquewhite" 
        onChangeText={(text) => setUsername(text)}
        keyboardType="ascii-capable"
      />
      <TextInput
        style={styles.input}
        onChangeText={(text => setPassword(text))}
        placeholderTextColor="antiquewhite" 
        placeholder="Password"
        secureTextEntry={true}
      />
      <Pressable
        style={styles.button}
        onPress={() => validateLogin()}>
        <Text
          style={styles.text}>Submit</Text>
      </Pressable>
      </SafeAreaView>
      <SafeAreaView style={styles.fillArea}/>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  input: {
    width:screenWidth - 100,
    paddingBottom: 10,
    alignItems: 'center',
    textAlign: 'center',
    borderBottomWidth: 2,
    borderColor: 'purple',
    color: 'white',
    fontSize: 18,
    borderRadius: 10,
    paddingHorizontal:25
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 10,
    elevation: 3,
    backgroundColor: 'purple',
    fontSize: 18,
  },
  text: {
    fontSize: 16,
    lineHeight: 21,
    fontWeight: 'bold',
    letterSpacing: 0.25,
    color: 'antiquewhite',
  },
  safeArea: {
    flex: 3,
    justifyContent: 'space-evenly',
    alignItems: 'center'
  },
  fillArea:{
    flex: 1
  }
});

export default LoginScreen;