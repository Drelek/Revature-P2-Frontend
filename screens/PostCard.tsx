import React, { useState, useEffect } from 'react';
import { View, Text, Image, StyleSheet, Pressable } from 'react-native'
import { Card } from 'react-native-elements'


const PostCard = (props: any) => {

    const [isLiked, setLikedState] = useState(false);
    const [numLikes, setNumLikes] = useState(0);

    useEffect(() => {
        //TODO
        //Pull the post array based on timestamp --> Iterate over array --> log number of likes

    })


    const fetchLikedState = () => {
        //TODO
    }

    const redirectToExtendedPostScreen = () => {
        //TODO

    }

    const renderNumOfComments = () => {
        const { comments } = props.item;
        if (comments.length) {
            return comments.length;
        } else {
            return '';
        }
    }

    const renderNumOfLikes = () => {
        const { likes } = props.item;
        if (likes.length) {
            return likes.length;
        } else {
            return '';
        }
    }

    const renderProfileImageOrDefault = () => {
        const { displayImg } = props.item;
        if (!displayImg) {
            return (
                <Image
                    source={require('../assets/images/illuminati.png')}
                    style={styles.defaultProfileImage}
                />
            )
        } else {
            return (
                <Image
                    source={{ uri: `${displayImg}` }}
                    style={styles.profileImage}
                />
            )
        }
    }

    const renderNotLikeOrLiked = () => {
        if (isLiked) {
            return (
                <Image
                    source={require('../assets/images/likeIcon.png')}
                    style={styles.heart}
                />)
        } else {
            return (<Image
                source={require('../assets/images/likedIcon.png')}
                style={styles.heart}
            />)
        }
    }


    return (
        <View
            style={styles.card}
        >

            <Card containerStyle={styles.cardActual}>

                <View
                    style={styles.containerHeadOfCard}
                >
                    <View style={styles.imageContainer}>
                        {renderProfileImageOrDefault()}
                    </View>


                    <View style={styles.nameContainer}>
                        <Text
                            style={styles.displayName}
                        >{props.item.displayName}</Text>
                        <Text
                            style={styles.username}
                        >{`@${props.item.userName}`}</Text>
                    </View>
                </View>

                <View style={styles.postContainer}>
                    <Text
                        style={styles.postBody}
                    >{props.item.postBody}</Text>
                </View>


                <View
                    style={styles.containerViewAlignIcons}
                >
                    <View style={styles.likesContainer}>
                        <Pressable onPress={() => setLikedState(!isLiked)}>
                            {renderNotLikeOrLiked()}
                        </Pressable>

                        <Text style={styles.likesText}>{renderNumOfLikes()}</Text>
                    </View>

                    <View style={styles.commentsContainer}>
                        <Pressable
                            onPress={() => redirectToExtendedPostScreen()}>
                            <Image
                                source={require('../assets/images/commentIcon.png')}
                                style={styles.comment}
                            />
                        </Pressable>
                        <Text style={styles.likesText}>
                            {renderNumOfComments()}
                        </Text>
                    </View>

                    <View style={styles.timeStampContainer}>
                        <Text style={styles.timestamp}>{props.item.timeStamp}</Text>
                    </View>

                </View>
            </Card>

        </View>
    )
}

export default PostCard;

const styles = StyleSheet.create({
    card: {
        padding: 10,
    },
    cardActual: {
        flex: 2,
        borderRadius: 10,
        borderColor: 'purple',
        borderWidth: 2,
        backgroundColor: 'rgb(33, 37, 41)',
    },
    containerHeadOfCard: {
        flexDirection: "row"
    },

    imageContainer: {
        flex: 1,
    },

    nameContainer: {
        flex: 2,
    },

    postContainer: {
        flex: 2,
        borderRadius: 20,
    },

    containerViewAlignIcons: {
        flex: 1,
        paddingTop: 10,
        flexDirection: "row",
        justifyContent: "space-around",
        borderTopWidth: 2,
        borderColor: "purple"
    },
    defaultProfileImage: {
        width: 80,
        height: 80,
        borderRadius: 100,
        borderWidth: 2,
        borderColor: 'purple',
        backgroundColor: 'purple'
    },
    profileImage: {
        width: 80,
        height: 80,
        borderRadius: 100,
        borderWidth: 2,
        borderColor: 'purple',
    },
    displayName: {
        fontSize: 22,
        color: "white",
        fontFamily: "BadScript"
    },
    username: {
        fontSize: 16,
        color: "white",
        fontFamily: "BadScript"
    },
    postBody: {
        color: "white",
        padding: 10,
        fontFamily: "Montserrat"
    },

    heart: {
        width: 25,
        height: 25

    },
    comment: {
        width: 25,
        height: 25,
    },

    timestamp: {
        color: "white",
        fontFamily: "Montserrat"
    },

    likesText: {
        color: "white",
        paddingLeft: 10,
        fontFamily: "Montserrat"
    },

    likesContainer: {
        flex: 1,
        flexDirection: "row",
    },

    commentsContainer: {
        flex: 1,
        flexDirection: "row"
    },

    timeStampContainer: {
    },
})