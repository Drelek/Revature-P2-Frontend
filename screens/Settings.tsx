import { View, Text, TextInput, StyleSheet, TouchableOpacity, SafeAreaView, KeyboardAvoidingView, Platform } from 'react-native'
import React, { useState } from 'react';
import { Card } from 'react-native-elements'
import { useDispatch, useSelector } from 'react-redux';
import { IAppState } from '../redux/store';
import axios from 'axios';
import { AppAction } from '../redux/actions';
import { IUser } from '../models/User';
import Toast from 'react-native-toast-message';

const SettingsScreens: React.FC = () => {

    let user = useSelector((store: IAppState) => store.user);
    const token = useSelector((state: IAppState) => state.auth.AccessToken);

    const dispatch = useDispatch();


    const [handle, setHandle] = useState(user?.displayName);
    const [profileImg, setProfileImage] = useState(user?.profileImg);
    const [working, setWorking] = useState(false);


    async function submitForm() {
        if (working) return;
        setWorking(true);

        const body = {
            'dataType': 'user',
            'dataKey': user?.userName,
            'displayName': handle,
            'profileImg': profileImg
        }
        const headers = {
            Authorization: token
        }
        try {
            const userResult = await axios.put('https://w822121nz1.execute-api.us-east-2.amazonaws.com/Prod/user/' + user?.userName, body, { headers });

            const newUser: IUser = {
                ...user,
                userName: userResult.data[0].dataKey,
                displayName: userResult.data[0].displayName,
                email: userResult.data[0].email,
                profileImg: userResult.data[0].profileImg,
            }

            console.log(newUser);

            dispatch({
                type: AppAction.UPDATE_USER,
                payload: { user: newUser }
            });

            Toast.show({
                type: "success",
                text1: "Submitted",
                text2: "Your new information has been submitted successfully"
            });
        }
        catch (err) {
            console.log(err.response)
            Toast.show({
                type: "error",
                text1: "Update Failed",
                text2: "Failed to update user data"
            })
        }
        setWorking(false);
    }

    return (
        <KeyboardAvoidingView behavior={Platform.OS === "ios" ? "padding" : null} style={styles.container}>
            <View style={styles.smallView}></View>
            <SafeAreaView style={styles.largeView}>
                <Card containerStyle={styles.cardActual}>
                    <View style={styles.titleContainer}>
                        <Text style={styles.text}> Profile Settings </Text>
                    </View>

                    <View style={styles.topForm}>

                    </View>
                    <View style={styles.form}>
                        <TextInput style={styles.input}
                            placeholderTextColor="white" placeholder="Display Name" onChangeText={(text) => setHandle(text)} />
                    </View>
                    <View style={styles.form}>
                        <TextInput style={styles.input}
                            placeholderTextColor="white" placeholder="Link to Profile Image" onChangeText={(text) => setProfileImage(text)} />
                    </View>
                    <View>
                        <TouchableOpacity style={[styles.button, working ? styles.working : styles.notWorking]} onPress={() => submitForm()}>
                            <Text style={styles.buttonText}>Submit</Text>
                        </TouchableOpacity>
                    </View>
                </Card>
            </SafeAreaView>
            <View style={styles.bottomView}></View>
        </KeyboardAvoidingView>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center'
    },
    smallView: {
        flex: 1,
    },
    bottomView: {
        flex: 1
    },
    largeView: {
        flex: 2,
        justifyContent: 'flex-end'
    },
    cardActual: {
        borderRadius: 10,
        borderColor: 'purple',
        borderWidth: 2,
        backgroundColor: 'rgb(33, 37, 41)',
        color: 'white',
    },

    topForm: {
        margin: 10
    },
    form: {
        margin: 10
    },

    titleContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
    },

    text: {
        fontSize: 25,
        color: "white",
        fontFamily: "Montserrat",
        alignItems: 'center',
        textAlign: 'center',
    },
    input: {
        fontFamily: "Montserrat",
        fontSize: 18,
        color: "white",
        borderBottomWidth: 2,
        borderColor: 'purple',
        paddingBottom: 10,
        alignItems: 'center',
        textAlign: 'center',
    },
    button: {
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 10,
        backgroundColor: 'purple',
        marginVertical: 10,
        marginHorizontal: 25
    },
    buttonText: {
        fontSize: 20,
        fontFamily: "Montserrat",
        color: "white",
    },
    notWorking: {
        backgroundColor: 'purple'
    },

    working: {
        backgroundColor: 'grey'
    }
})

export default SettingsScreens;

