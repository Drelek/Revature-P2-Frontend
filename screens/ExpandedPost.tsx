import React, { useState, useEffect } from 'react';
import { View, FlatList, Pressable, Text, StyleSheet, Image, KeyboardAvoidingView, Platform } from 'react-native';
import IndividualComment from './IndividualComment';
import AddComment from './AddComment';
import { Card } from 'react-native-elements'
import axios from 'axios';
import { useSelector } from 'react-redux';
import { IAppState } from '../redux/store';

const ExpandedPost: React.FC = (props: any) => {

    const [isLiked, setLikedState] = useState(false);
    const [commentList, setCommentList] = useState([]);
    const token = useSelector((state: IAppState) => state.auth.AccessToken);


    const grabCommentsActual = async () => {
        await axios.get(`https://w822121nz1.execute-api.us-east-2.amazonaws.com/Prod/post/${props.route.params.dataKey}`, {
            headers: {
                Authorization: token
            }
        }).then(resp => {
            setCommentList(resp.data[0].comments);
            console.log(resp.data);
        })

    }

    const renderNumOfLikes = (likes: number[]) => {
        if (likes.length) {
            return likes.length;
        } else {
            return '';
        }
    }

    const renderProfileImageOrDefault = (displayImg: string) => {
        if (!displayImg) {
            return (
                <Image
                    source={require('../assets/images/illuminati.png')}
                    style={styles.defaultProfileImage}
                />
            )
        } else {
            return (
                <Image
                    source={{ uri: `${displayImg}` }}
                    style={styles.profileImage}
                />
            )
        }
    }

    const renderNotLikeOrLiked = () => {
        if (isLiked) {
            return (
                <Image
                    source={require('../assets/images/likeIcon.png')}
                    style={styles.heart}
                />)
        } else {
            return (<Image
                source={require('../assets/images/likedIcon.png')}
                style={styles.heart}
            />)
        }
    }

    useEffect(() => {
        grabCommentsActual();
    }, []);

    const { displayName, displayImg, userName, postBody, likes } = props.route.params
    const timeStamp = new Date(Number(props.route.params.dataKey)).toLocaleTimeString() + ' ' + new Date(Number(props.route.params.dataKey)).toLocaleDateString()

    const renderSinglePost = () => {

        return (
            <View style={styles.postContainer}>

                <Card containerStyle={styles.cardActual}>

                    <View style={styles.headerContainer}>
                        <View>{renderProfileImageOrDefault(displayImg)}</View>

                        <View style={styles.infoContainer}>
                            <View><Text style={styles.displayName}>{displayName}</Text></View>
                            <View><Text style={styles.userName}>{`@${userName}`}</Text></View>
                        </View>

                    </View>

                    <View style={styles.bodyContainer}><Text style={styles.postBody}>{postBody}</Text></View>

                    <View style={styles.footerContainer}>
                        <View style={styles.likesContainer}>
                            <Pressable onPress={() => setLikedState(!isLiked)}>
                                {renderNotLikeOrLiked()}
                            </Pressable>

                            <Text style={styles.likesText}>{renderNumOfLikes(likes)}</Text>
                        </View>

                        <View style={styles.timeStampContainer}><Text style={styles.timeStamp}>{timeStamp}</Text></View>
                    </View>

                </Card>
            </View>
        )
    }

    return (

        <KeyboardAvoidingView style={styles.container} behavior={Platform.OS === "ios" ? "padding" : "null"} keyboardVerticalOffset={Platform.select({ ios: 64 })}>
            {/* <Pressable onPress={ () => mergePostCommentData()}>
                    <Text style={{color:"white"}}>HELLO</Text>
                </Pressable>
                <Pressable onPress={ () => console.log(commentList)}>
                    <Text style={{color:"white"}}>EAZYMONEY</Text>
                </Pressable> */}
            {/* <View style={styles.inner}> */}
            <View style={styles.commentsContainer}>

                <FlatList
                    data={commentList}
                    ListHeaderComponent={() => renderSinglePost()}
                    renderItem={({ item }) => <IndividualComment item={item}></IndividualComment>}
                    keyExtractor={(item, index) => index.toString()}
                />

            </View>
            {/* <View style={styles.addCommentContainer}> */}
            <AddComment text={"Leave a Reply"} timeStamp={props.route.params.dataKey}></AddComment>
            {/* </View> */}
            {/* </View> */}
        </KeyboardAvoidingView>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },

    inner: {
        flex: 1,
        justifyContent: "flex-end",
    },

    postContainer: {
        flex: 2,
        flexDirection: 'column'
    },

    cardActual: {
        borderWidth: 5,
        borderColor: "purple",
        borderRadius: 10,
        backgroundColor: "rgb(33, 37, 41)"
    },

    headerContainer: {
        flexDirection: "row",
        marginBottom: 10
    },

    profileImage: {
        height: 80,
        width: 80,
        borderRadius: 100,
        borderWidth: 2,
        borderColor: "purple",
        marginRight: 15
    },

    infoContainer: {
        flex: 2,
        flexDirection: "column",
        justifyContent: "center"
    },

    bodyContainer: {
        flex: 1
    },

    footerContainer: {
        flex: 1,
        paddingTop: 10,
        flexDirection: "row",
        justifyContent: "space-around",
        borderTopWidth: 2,
        borderColor: "purple",
    },

    likesContainer: {
        flex: 1,
        flexDirection: "row",
    },

    blank: {
        flex: 1,
    },

    timeStampContainer: {
        flex: 1,
        flexDirection: "row",
        justifyContent: "flex-end"
    },

    commentsContainer: {
        flex: 8,
        marginBottom: 10
    },

    addCommentContainer: {
        flex: 1,
        justifyContent: "flex-end"
    },

    displayName: {
        color: "white",
        fontFamily: "BadScript",
        fontSize: 20
    },

    userName: {
        color: "white",
        fontFamily: "BadScript"
    },

    postBody: {
        color: "white",
        marginBottom: 10,
        fontSize: 16,
        fontFamily: "Montserrat"
    },

    defaultProfileImage: {
        marginTop: 5,
        width: 90,
        height: 90,
        borderRadius: 100,
        borderWidth: 2,
        borderColor: 'purple',
    },

    heart: {
        width: 25,
        height: 25
    },

    timeStamp: {
        color: "white",
    },

    likesText: {
        color: "white",
        marginLeft: 10
    }
})
export default ExpandedPost;