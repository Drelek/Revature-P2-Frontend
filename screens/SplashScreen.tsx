import React, { useState } from 'react';
import { StyleSheet, Text, SafeAreaView, Pressable, KeyboardAvoidingView } from 'react-native';
import AnimatedTypeWriter from 'react-native-animated-typewriter';
import { screenWidth } from '../constants/Layout';
import LoginScreen from './LoginScreen';
import SignUpScreen from './SignUpScreen';

const SplashScreen: React.FC = (props: any) => {

  const [userSession, setUserSession] = useState({
    session: "login"
  });


  const renderSession = () => {
    const session = userSession.session;
    if (session === "login") {
      return (
        <LoginScreen />
      )
    } else if (session === "sign-up") {
      return (
        <SignUpScreen />
      )
    }
  }

  const renderSessionButton = () => {
    const session = userSession.session;
    if (session === "login") {
      return <Pressable
        onPress={() => setUserSession({ session: "sign-up" })}>
        <Text
          style={styles.text}>Sign Up</Text>
      </Pressable>
    } else if (session === "sign-up") {
      return <Pressable
        onPress={() => setUserSession({ session: "login" })}>
        <Text
          style={styles.text}>Login</Text>
      </Pressable>
    }
  }
  function welcomeMessage() {
    const message: string = "Welcome to Bohemian Grove, the message app for those in the know.";
    return <AnimatedTypeWriter containerStyle={styles.message} textStyle={styles.text} text={`${message}`} />
  }

  return (
    <SafeAreaView style={styles.container}>
      <SafeAreaView style={styles.smallView} />

      <SafeAreaView style={styles.smallView}>
        {welcomeMessage()}
      </SafeAreaView>

      <SafeAreaView style={styles.largeView}>
        {renderSession()}
        <SafeAreaView>{renderSessionButton()}</SafeAreaView>
      </SafeAreaView>

      <SafeAreaView style={styles.smallView} />

    </SafeAreaView>
  )
}

export default SplashScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    zIndex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },

  smallView: {
    flex: 1,
    alignItems: 'center',
    paddingHorizontal: 15,
    backgroundColor: "transparent",
    borderColor: "purple"
  },

  largeView: {
    flex: 3,
    backgroundColor: 'rgb(33, 37, 41)',
    borderRadius: 10,
    borderWidth: 4,
    borderColor: 'purple',
    width: screenWidth - 20,
  },

  text: {
    color: "white",
    fontSize: 20,
    backgroundColor: "transparent",
    // fontFamily: "Montserrat",
  },

  message: {
    color: 'white',
    fontSize: 18,
    backgroundColor: 'rgb(33, 37, 41)',
    textAlign: "center",
    borderColor: 'purple',
    borderWidth: 4,
    borderRadius: 10,
    paddingHorizontal: 15,
    // fontFamily: "BadScript-Regular"
  }
})
