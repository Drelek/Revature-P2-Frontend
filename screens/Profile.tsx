import * as React from 'react';
import { Text, TextInput, View, StyleSheet, Image, SafeAreaView, Pressable, FlatList } from 'react-native';
import { Card } from 'react-native-elements';
import { useState } from 'react';
import PostCard from './PostCard';
import ProfileImgPlaceholder from '../assets/images/profile-img-placeholder.png'
import { ScrollView } from 'react-native-gesture-handler';


const Profile = () => {

    const [postCards, setPostCards] = useState([
        {
            displayImg: "null",
            displayName: "my name is",
            userName: "name",
            postBody: "I know the truth",
            likes: "hi",
            timeStamp: "12312",
            comments: "hello"
        },
        {
            displayImg: "null",
            displayName: "my name is",
            userName: "name",
            postBody: "I know the truth",
            likes: "hi",
            timeStamp: "12312",
            comments: "hello"
        }
    ]);

    const addFollower = () => {
        //TODO
        //Push follower to following array on user object
    }

    

    return(
        <View 
            style= {styles.outerContainer}
        >   
            <ScrollView>
            <View
                style={styles.innerContainer}
            >

               

                <Card containerStyle={styles.profileCard}>
                    <View
                        style={{flexDirection: "row"}}
                    > 
                        <Image
                            source={require('../assets/images/profile-img-placeholder.png')}
                            style={styles.image}
                        />

                        <View>
                            <Text 
                                style={styles.displayname}
                            >HolyGuack</Text>
                            <Text
                                style={styles.username}
                            >@guackholy</Text>
                            <Text
                                style={styles.email}
                            >Email Placeholder</Text>     
                        </View>

                        <View>
                        <Pressable
                            onPress= {() => addFollower()}
                        >
                            <Image 
                                style={styles.followerIcon}
                               source={require('../assets/images/follower-icon.png')}
                          />
                         </Pressable>
                          </View>
                    </View>
                </Card>

                
            </View>

            <SafeAreaView
            
            >
                
                <FlatList 
                    data={postCards}
                    renderItem={({item}) => 
                        <PostCard item={item}></PostCard>
                    }    
                />

            </SafeAreaView>

            </ScrollView>

   
        </View>
    )
}

export default Profile;

const styles = StyleSheet.create({
    outerContainer: {
        flexDirection: "column",
        alignItems: "center",
    },
    innerContainer: {
        flexDirection: "column",
        alignItems: "center",
        
    },
    profileCard: {
        flex:1, 
        borderTopLeftRadius:20, 
        borderColor: 'plum', 
        borderWidth: 1,
        borderTopRightRadius:20,
        borderBottomRightRadius: 20,
        borderBottomLeftRadius: 20,
        backgroundColor: 'rgb(33, 37, 41)',
        width: 290,
        height: 110
    },
    image: {
        width: 80,
        height: 80,
        borderRadius: 25,
    },
    listOfPosts: {
        
    },
    displayname: {
        fontWeight: "bold",
        fontSize: 25,
        color: "white",
        paddingLeft: 15
    },
    username: {
        fontSize: 18,
        color: "white",
        paddingLeft: 15
    },
    email: {
        fontSize: 18,
        color: "white",
        paddingLeft: 15
    },
    welcomeMessage: {
        fontSize: 30, 
        fontWeight: "bold", 
        color: "white", 
        fontStyle: "italic"
    },
    followerIcon: {
        width: 50,
        height: 60
    }
})