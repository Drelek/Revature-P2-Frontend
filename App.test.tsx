import Enzyme from 'enzyme'
import React from 'react';
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { reducers } from './redux/session_reducers';
import SplashScreen from './screens/SplashScreen'
import Settings from './screens/Settings'
import Profile from './screens/Profile'
import PostCard from './screens/PostCard'
import LoginScreen from './screens/LoginScreen'
import HomeFeedScreen from './screens/HomeFeed'
import ExpandedPost from './screens/ExpandedPost';
import AddComment from './screens/AddComment';
import handleCanvas from './components/canvas';
import FeedPicker from './components/feedPicker';
import FollowIcon from './components/followIcon';
import GlobalEye from './components/globalEye';
import Logo from './components/logo';
import { Menu } from 'react-native-paper';
import MenuIcon from './components/menuIcon';






const store = createStore(reducers);
Enzyme.configure({ adapter: new Adapter() });

/*
**  Screen Render Testing
**
*/
describe('SplashScreen', () => {
    it('renders without crashing', () => {
        const component = shallow(<SplashScreen />);
        expect(component).toMatchSnapshot();
    });
})


describe('Settings', () => {
    it('renders without crashing', () => {
        const component = shallow(<Settings />);
        expect(component).toMatchSnapshot();
    });
})

describe('Profile', () => {
    it('renders without crashing', () => {
        const component = shallow(<Profile />);
        expect(component).toMatchSnapshot();
    });
})



describe('LoginScreen', () => {
    it('renders without crashing', () => {
        const component = shallow(<Provider store={store}><LoginScreen /></Provider>);
        expect(component).toMatchSnapshot();
    });
})

describe('HomeFeedScreen', () => {
    it('renders without crashing', () => {
        const component = shallow(<HomeFeedScreen />);
        expect(component).toMatchSnapshot();
    });
})

describe('Expanded Post', () => {
    it('renders without crashing', () => {
        const component = shallow(<Provider store={store}><ExpandedPost /></Provider>);
        expect(component).toMatchSnapshot();
    });
})

describe('AddComment', () => {
    it('renders without crashing', () => {
        const component = shallow(<AddComment />);
        expect(component).toMatchSnapshot();
    });
})

/*
**  Component Render Testing
**
*/

describe('Canvas comp', () => {
    it('renders without crashing', () => {
        const component = shallow(<handleCanvas />);
        expect(component).toMatchSnapshot();
    });
})


describe('FollowIcon comp', () => {
    it('renders without crashing', () => {
        const component = shallow(<FollowIcon />);
        expect(component).toMatchSnapshot();
    });
})

describe('GlobalEye comp', () => {
    it('renders without crashing', () => {
        const component = shallow(<GlobalEye />);
        expect(component).toMatchSnapshot();
    });
})

describe('Logo comp', () => {
    it('renders without crashing', () => {
        const component = shallow(<Logo />);
        expect(component).toMatchSnapshot();
    });
})

