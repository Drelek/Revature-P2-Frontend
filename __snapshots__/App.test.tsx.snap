// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddComment renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Canvas comp renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Expanded Post renders without crashing 1`] = `ShallowWrapper {}`;

exports[`FeedPicker comp renders without crashing 1`] = `ShallowWrapper {}`;

exports[`FollowIcon comp renders without crashing 1`] = `ShallowWrapper {}`;

exports[`GlobalEye comp renders without crashing 1`] = `ShallowWrapper {}`;

exports[`HomeFeedScreen renders without crashing 1`] = `ShallowWrapper {}`;

exports[`LoginScreen renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Logo comp renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Profile renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Settings renders without crashing 1`] = `ShallowWrapper {}`;

exports[`SignInUpComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`SplashScreen renders without crashing 1`] = `ShallowWrapper {}`;
